packages/block-library/src/embed/embed-preview.js

```diff
@@ -17,57 +17,93 @@ import classnames from 'classnames/dedupe';
-	const sandboxClassnames = classnames( type, className, 'wp-block-embed__wrapper' );
+	hideOverlay() {
+		// This is called onMouseUp on the overlay. We can't respond to the `isSelected` prop
+		// changing, because that happens on mouse down, and the overlay immediately disappears,
+		// and the mouse event can end up in the preview content. We can't use onClick on
+		// the overlay to hide it either, because then Gutenberg misses the mouseup event, and
+		// thinks we're multi-selecting blocks.
+		this.setState( { interactive: true } );
+	}
 
-	const embedWrapper = 'wp-embed' === type ? (
-		<WpEmbedPreview
-			html={ html }
-		/>
-	) : (
-		<div className="wp-block-embed__wrapper">
-			<SandBox
+	static getDerivedStateFromProps( nextProps ) {
+		if ( ! nextProps.isSelected ) {
+			// We only want to change this when the block is not selected, because changing it when
+			// the block becomes selected makes the overlap disappear too early. Hiding the overlay
+			// happens on mouseup when the overlay is clicked.
+			return { interactive: false };
+		}
+	}
+
+	render() {
+		const { preview, url, type, caption, onCaptionChange, isSelected, className, icon, label } = this.props;
+		const { scripts } = preview;
+		const { interactive } = this.state;
+
+		const html = 'photo' === type ? getPhotoHtml( preview ) : preview.html;
+		const parsedUrl = parse( url );
+		const cannotPreview = includes( HOSTS_NO_PREVIEWS, parsedUrl.host.replace( /^www\./, '' ) );
+		// translators: %s: host providing embed content e.g: www.youtube.com
+		const iframeTitle = sprintf( __( 'Embedded content from %s' ), parsedUrl.host );
+		const sandboxClassnames = classnames( type, className, 'wp-block-embed__wrapper' );
+
+		/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
+		const embedWrapper = 'wp-embed' === type ? (
+			<WpEmbedPreview
 				html={ html }
-				scripts={ scripts }
-				title={ iframeTitle }
-				type={ sandboxClassnames }
 			/>
-		</div>
-	);
-
-	return (
-		<figure className={ classnames( className, 'wp-block-embed', { 'is-type-video': 'video' === type } ) }>
-			{ ( cannotPreview ) ? (
-				<Placeholder icon={ <BlockIcon icon={ icon } showColors /> } label={ label }>
-					<p className="components-placeholder__error"><a href={ url }>{ url }</a></p>
-					<p className="components-placeholder__error">{ __( 'Previews for this are unavailable in the editor, sorry!' ) }</p>
-				</Placeholder>
-			) : embedWrapper }
-			{ ( ! RichText.isEmpty( caption ) || isSelected ) && (
-				<RichText
-					tagName="figcaption"
-					placeholder={ __( 'Write caption…' ) }
-					value={ caption }
-					onChange={ onCaptionChange }
-					inlineToolbar
+		) : (
+			<div className="wp-block-embed__wrapper">
+				<SandBox
+					html={ html }
+					scripts={ scripts }
+					title={ iframeTitle }
+					type={ sandboxClassnames }
 				/>
-			) }
-		</figure>
-	);
+				{ ! interactive && <div
+					role="dialog"
```

Comment:
this div is not an ARIA dialog, doesn't have a particular semantic meaning nor can be focused so this role needs to be removed